pulp_rpm/app/tasks/publishing.py

```diff
@@ -156,28 +161,48 @@ def populate(self):
             self.handle_sub_repos(distribution_tree)
 
         all_content = main_content
-        for name, content in self.sub_repos:
+        for name, content, checksums in self.sub_repos:
             os.mkdir(name)
             setattr(self, f"{name}_content", content)
             setattr(self, f"{name}_repomdrecords", self.prepare_metadata_files(content, name))
             all_content |= content
         self.publish_artifacts(all_content)
 
 
-def publish(repository_version_pk, metadata_signing_service=None):
+def get_checksum(cr, name, checksums):
+    """
+    Get checksum algorithm for publishing metadata.
+
+    """
+    original = checksums.get("original")
+    metadata = checksums.get("metadata")
+    package = checksums.get("package")
+    checksum_type = original.get(name, "SHA256")
+
+    if metadata:
+        checksum_type = metadata
+    if package and name in PACKAGE_REPODATA:
+        checksum_type = package
+
+    return getattr(cr, checksum_type, 5)
```

Comment:
No magic numbers please.
What is 5? I would guess it's a createrepo_c code for sha256. If so, please, take the constant from createrepo_c: `cr.SHA256`.